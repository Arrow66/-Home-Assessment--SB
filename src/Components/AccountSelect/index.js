import React,{ useContext } from "react"
import StyledSelect from "../Select"
import { accountsContext } from "../../context"
import { setSelectedAccountAction } from "../../context/utils"


const { ValueContext, ValueDispatchContext } = accountsContext

const AccountSelect = () => {
    const dispatch = useContext(ValueDispatchContext);
    const { selectedAccount, accounts } = useContext(ValueContext);

    const options = Object.entries(accounts).map(([key, value]) => {
        return {
            value: key,
            label: value.accountName,
            key
        }
    })

    return <StyledSelect options={options} onChange={(e) => {
        setSelectedAccountAction(dispatch, e.target.value)
    }}
        value={selectedAccount}
    />

}

export default AccountSelect