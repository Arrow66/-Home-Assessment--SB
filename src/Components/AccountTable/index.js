import React, { useContext } from "react"
import Table from "../Table"
import { accountsContext } from "../../context";


const { ValueContext } = accountsContext


const AccountTable = () => {
    const { selectedAccount, accounts } = useContext(ValueContext);
    const tableData = accounts[selectedAccount];
    const { docs } = tableData || {}

    const TableRender = docs&&docs.map(({ document_type, is_paper_flag }) => (
        <Table.Row>
            <Table.Cell>{document_type}</Table.Cell>
            <Table.Cell>{is_paper_flag ? "Paper" : "Paperless"}</Table.Cell>
        </Table.Row>
    ))||[]

    return (
        <Table>
            <Table.Header>
                <Table.Column> Document Type</Table.Column>
                <Table.Column>Delivery Method</Table.Column>
            </Table.Header>
            <Table.Body>
                {
                    TableRender
                }
            </Table.Body>
        </Table>
    )

}

export default AccountTable;