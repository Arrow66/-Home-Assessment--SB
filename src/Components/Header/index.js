import React from "react";

const Header = () => (
    <header className="App-header" role="banner">
        <h1 className="App-title" id="main-heading">Demo Account and Document Preference Delivery</h1>
    </header>
);

export default Header;
