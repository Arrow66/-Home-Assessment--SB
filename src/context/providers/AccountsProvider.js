import React, {useEffect, createContext,useReducer } from "react";
import mockData, { defaultAccount } from "../../../static/mockData";
import { fetchDataSuccess, setDefaultAccount } from "../utils";
import reducer from "../reducers";



export const ValueContext = createContext(null);
export const ValueDispatchContext = createContext(null);

const {accountsReducer,initialState} = reducer;

 const AccountsProvider  = ({children})=>{

    useEffect(()=>{
        fetchDataSuccess(dispatch,mockData)
        setDefaultAccount(dispatch,defaultAccount)
    },[])
    

    const [accounts,dispatch] = useReducer(accountsReducer,initialState);

    return (
        <ValueContext.Provider value={accounts}>
            <ValueDispatchContext.Provider value={dispatch}>
                {children}
            </ValueDispatchContext.Provider>
        </ValueContext.Provider>
    )

}


export default AccountsProvider;
