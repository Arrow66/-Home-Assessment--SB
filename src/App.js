import React,{lazy,Suspense} from 'react'
import '../static/css/App.css'
import Providers from './context/providers';
import Header from './Components/Header';
import Box from "./Components/Box";
const  AccountSelect  = lazy(()=>import('./Components/AccountSelect'));
const  AccountTable  =  lazy(()=>import('./Components/AccountTable'));

const { AccountsProvider } = Providers;

const App = () => {

    return (
        <AccountsProvider>
            <div className='App'>
                <Header />
                <main role='main'>
                    <div className='container'>
                        <Box role='region'>
                            <Suspense fallback={<div>Loading...</div>}>
                            <AccountSelect />
                            </Suspense>
                        </Box>
                        <Box role="region" aria-label='Account Table'>
                        <Suspense fallback={<div>Loading...</div>}>
                            <AccountTable />
                        </Suspense>
                        </Box>
                    </div>
                </main>
            </div>
        </AccountsProvider>
    )
}


export default App;
