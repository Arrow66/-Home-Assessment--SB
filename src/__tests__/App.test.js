import { shallow,mount } from "enzyme"
import App from "../App"
import Header from "../Components/Header"
import AccountSelect from "../Components/AccountSelect"
import AccountTable from "../Components/AccountTable"
import renderer from "react-test-renderer"
import AccountsProvider from "../context/providers/AccountsProvider"
import React from "react"
import Box from "../Components/Box"



describe('App', () => {
    it("render without crashing", () => {
        shallow(<App />);
    })
    it('should match snapshot', () => {
        const tree = renderer
          .create(
            <App />
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    it('renders the Header component', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find(Header)).toHaveLength(1);
      });
      it('renders the AccountsProvider and wraps the entire component', () => {
        const wrapper = mount(<App />);
        expect(wrapper.find(AccountsProvider)).toHaveLength(1);
        expect(wrapper.find(AccountsProvider).children()).toHaveLength(1);
      });
      it('renders the main container and boxes correctly', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('main')).toHaveLength(1);
        expect(wrapper.find('.container')).toHaveLength(1);
        expect(wrapper.find(Box)).toHaveLength(2);
      });
      
})