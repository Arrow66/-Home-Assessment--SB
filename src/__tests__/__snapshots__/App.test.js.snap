// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
    role="banner"
  >
    <h1
      className="App-title"
      id="main-heading"
    >
      Demo Account and Document Preference Delivery
    </h1>
  </header>
  <main
    role="main"
  >
    <div
      className="container"
    >
      <div
        className="sc-beySPh kIPIVA"
        role="region"
      >
        <div>
          Loading...
        </div>
      </div>
      <div
        aria-label="Account Table"
        className="sc-beySPh kIPIVA"
        role="region"
      >
        <div>
          Loading...
        </div>
      </div>
    </div>
  </main>
</div>
`;
